src/session/session_prefetch.c

```diff
@@ -10,34 +10,26 @@
 
 /*
  * __wt_session_prefetch_check --
- *     Check to see whether cursors owned by this session might benefit from doing pre-fetch
- *
- * FIXME-WT-11758 Change this function so that when pre-fetching is enabled on the session level,
- *     all cursors associated with that session with automatically perform pre-fetching. Use the
- *     session-level configuration once it is introduced.
+ *     Check if pre-fetching work should be performed for a given ref.
  */
 bool
 __wt_session_prefetch_check(WT_SESSION_IMPL *session, WT_REF *ref)
 {
-    /* Internal threads should not be configured to do pre-fetching. */
-    if (!S2C(session)->prefetch_auto_on || F_ISSET(session, WT_SESSION_INTERNAL))
-        return (false);
-
     if (S2C(session)->prefetch_queue_count > WT_MAX_PREFETCH_QUEUE)
```

Comment:
It seems a bit surprising that sessions that aren't configured for prefetch can increment statistics related to prefetch.
i.e: I'd be inclined to move the `WT_SESSION_PREFETCH` check above this code that bumps a statistic.